Add plugin documentation (#46)
mathildebrosseau/transition_qgis: README.md

# Transition QGIS
Transition is a QGIS plugin designed to interact with the transit planning application Transition.\
Before using this plugin, users need to install the Transition python library that this plugin depends on in order to interact with the application. To install it, use the following command :
```bash
pip install transition-lib
```
## Usage (end user)

> Install `Transition` through the QGIS plugin manager.

Reviewer: avec une photo aussi peut-être ?

When the plugin is open for the first time, users are prompt to enter their login credentials in order to be authentified with the Transition application.

![alt text](docs/screenshots/login_prompt.png)

> The prompt also asks users for the server URL that is to be used. By default, it is set to `http://localhost:8080` but can be edited by users. The login gets an authentication token from the Transition application. When the token expires, the user will be prompted to login again to get a new one.

Reviewer: ```suggestion
Users also have to provide the URL to the Transition server URL to be used. The default value is `http://localhost:8080`, but can it be edited. 

The login process retrieves an authentication token from the Transition application. When the token expires, the user will be prompted to login again to obtain a new one.
```

Author: "Users also have to provide the URL to the Transition server URL to be used"
je pense que la fin de la phrase est pas vraiment bonne ?

Reviewer: au pire on enlève le "to be used" ? je pense ça se dit bien quand même

> After a successful login, users have access to the plugin, which is a dockwidget.\

Reviewer: ```suggestion
After a successful login, the plugin appears as a dockwidget.\
```

> The dockwidget contains multiple tabs which are described in the following.

Reviewer: ```suggestion
The dockwidget features multiple tabs, each of which is outlined below.
```

### Home page

![alt text](docs/screenshots/home_page.png)

The home page offers the following functionalities :
- Get all the paths currently loaded in the Transition application.
- Get all the nodes currently loaded in the Transition application.
The results for each request will be displayed as a layer in QGIS.

### Routing page
![alt text](docs/screenshots/routing_page.png)

The routing page allows users to send a calculation request to the Transition application in order to receive a route. Users can choose transit modes, origin and destination points, as well as different parameters for transit, such as departure and waiting time.\
The result will be displayed as a group layer containing a separate layer for each mode. 

> For transit, if the **Calculate with alternatives** option is checked, multiple alternatives can be returned. When that is the case, the alternative routes will be displayed in a sub-group.

Reviewer: Ajouter l'explication et la photo pour le Route name

> For transit, if the **Calculate with alternatives** option is checked, multiple alternatives can be returned. When that is the case, the alternative routes will be displayed in a sub-group.

Reviewer: Expliquer que les résultats se font écraser quand on redemande une route sans spécifier le nom

### Accessibility map page
![alt text](docs/screenshots/accessibility_map_page.png)

The accessibility map page allows users to send a calculation request to the Transition application in order to receive an accessibility map. Users can choose the origin point as well as different parameters such as the departure and waiting time.\

> If the **Separate polygons into different layers** option is checked, the result will be a group layer containing a separate layer for each polygon. Otherwise, the result will just be one layer containing all the polygons.

Reviewer: Ajouter l'explication et la photo pour le Accessibility map name + Expliquer que les résultats se font écraser quand on redemande une route sans spécifier le nom

### Settings page
![alt text](docs/screenshots/settings_page.png)

The settings page contains read-only fields that display the connected user and the server URL currently being used. This mainly allows users to validate if the current URL is the desired one or if it needs to be changed.\

> The page also allows users to disconnect from the plugin.  

Reviewer: ```suggestion
The page also allows users to log out from the plugin.  
```

## Contribute

### Clone the repository and install dependencies

> To contribute to the Transition QGIS plugin, you can clone this reponsitory in `~/.local/share/QGIS/QGIS3/profiles/default/python/plugins/`. The plugin should then be added to your installed plugins and can be visible in the plugin bar.

Reviewer: Non il faut que l'utilisateur fasse qqch dans le plugin manager pour que le plugin s'affiche une fois que le dossier est cloné messemble

Author: je pense pas, de mémoire c'était juste une ligne de code que tu avais ajouté pour le voir puis elle est déjà pushed

Reviewer: faudrait retourner voir nos tutoriels lol

> ![alt text](docs/screenshots/plugin_icons.png)

Reviewer: ajouter cette tite image là au début aussi quand on explique comment ajouter le plugin!

In order to contribute to the project, you need to install the project dependencies. To do that, run the command :
```bash
pip install -r requirements.txt
``` 
You can then add or edit the code and see it reflected in QGIS. It is recommended to install the `plugin_reloader` plugin in QGIS in order to be able to easily reload the Transition plugin when files have been edited.

### Translate the plugin

> When new code is added, the fileds that are going to be displayed in the QGIS UI need to be translated to french. In order to allow that, you first need to install QT Linguist.

Reviewer: ```suggestion
When new code is added, the fields that are going to be displayed in the QGIS UI need to be translated to french. In order to allow that, you first need to install QT Linguist.
```

> The fields that are to be translated need to be wrapped in a `self.tr` clause. For example :

Reviewer: ```suggestion
All text visible by the user needs to be translated. To allow this, they need to be wrapped in a `self.tr` clause. For example :
```

```python
CustomLabel(self.tr("Departure or arrival time"))
```
The files that need to be translated need to be specified in the `i18n/transition_qgis.pro`. Once all the files are added to the .pro file, you need to run the followind command in the `i18n` directory :
```bash
pylupdate5 transition_qgis.pro
```

> After that, you can open the `transition_qgis_fr.ts` file in QT Linguist and write the translation of each field. Finally, once that is done, you need to run the following command in the `i18n` directory :

Reviewer: ```suggestion
After that, you can open the `transition_qgis_fr.ts` file in QT Linguist, write the translation of each field and save your changes. Once that is done, run the following command in the `i18n` directory :
```
